src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/ApplicationsController.kt

```diff
@@ -134,15 +136,28 @@ class ApplicationsController(
     return ResponseEntity.ok(getPersonDetailAndTransform(updatedApplication, user))
   }
 
-  private fun getPersonDetailAndTransformToSummary(
-    application: uk.gov.justice.digital
-    .hmpps.approvedpremisesapi.jpa.entity.Cas2ApplicationSummary,
-    user: NomisUserEntity,
-  ):
-    ApplicationSummary {
-    val personInfo = offenderService.getInfoForPersonOrThrowInternalServerError(application.getCrn())
+  private fun getPeopleDetailsAndTransformToSummaries(applications: List<Cas2ApplicationSummary>, user: NomisUserEntity): List<ApplicationSummary> {
+    val peopleInfo = offenderService.getOffenderSummariesByCrns(
+      applications.map { it.getCrn() }.toSet(),
+    )
+
+    val summaries = mutableListOf<ApplicationSummary>()
+
+    for (application in applications) {
+      val crn = application.getCrn()
+      val offenderDetailSummary = peopleInfo[crn] ?: throw NotFoundProblem(crn, "No offender details found")
```

Comment:
I think we can follow our previously established pattern of not throwing an error if the offender details haven't been found https://github.com/ministryofjustice/hmpps-approved-premises-api/commit/5360a6cd22ffd663df506dbcd4a59603a62bd72f

I don't think we want the whole request to error if the offender details haven't been found, but just to return with placeholder copy - if it's because of upstream api being down, that will be fixed shortly, if it's because person is genuinely not found then the user still needs to be able to access the application.

Reply:
also means we will have to update the test `Get list of applications returns 500 when a person cannot be found` because we no longer want to return a 500